src/setupWorker/start/createStart.ts

```diff
@@ -61,6 +64,22 @@ export const createStart = (context: SetupWorkerInternalContext) => {
       const [worker, registration] = instance
 
       if (!worker) {
+        if (options?.findWorker) {
+          console.error(`\
+[MSW] Failed to locate the Service Worker registration using a custom "findWorker" predicate.
+
+Please ensure that the custom predicate properly locates the Service Worker registration at "${resolvedOptions.serviceWorker.url})".
+More details: https://mswjs.io/docs/api/setup-worker/start#findworker
+`)
+        } else {
+          console.error(`\
+[MSW] Failed to locate the Service Worker registration.
+
+This most likely means that the worker script URL ("${resolvedOptions.serviceWorker.url}") cannot resolve against the actual public hostname (${location.host}). This may happen if your application runs behind a proxy, or has a dynamic hostname.
```

Comment:
@msutkowski may I please ask you to check if this error message contains proper hostname information from the #335 (if you have the reproduction repository)? I expect this error to display the worker script URL and the current proxied hostname, suggesting the difference.

I know the worker script URL is not absolute at this point, but it should be sufficient.

Reply:
Alright, looks like we're good to go here. If someone else cares to validate (@benmonro ?), please feel free.

Without specifying any options and just calling `worker.start()` on the example project: 

![image](https://user-images.githubusercontent.com/784953/91459946-1bc81280-e83c-11ea-8819-dc146ca09f0e.png)

Confirmed works as expected when passing in:

```js
 worker.start({
    findWorker: (scriptUrl, _mockServiceWorkerUrl) => scriptURL.includes('mockServiceWorker')
  })
```

![image](https://user-images.githubusercontent.com/784953/91460756-0bfcfe00-e83d-11ea-923f-7ab69bb7b1c6.png)

Reply:
And here's the full error with specifying a worker serviceWorker.url that doesn't match due to proxy:

![image](https://user-images.githubusercontent.com/784953/91461873-6480cb00-e83e-11ea-92d5-9cf4e4a346ab.png)

Reply:
Thank you for verifying this, @msutkowski! Happy to see that error message being as expected.